brut.j.util/src/main/java/brut/util/OSDetection.java

```diff
@@ -39,4 +41,26 @@ public static boolean is64Bit() {
     public static String returnOS() {
         return OS;
     }
+    public static boolean isAndroid() {
+        return new File(android_linker).exists();
+    }
+    public static String android_arch() {     
+        if(new File(android_linker).exists()){
+            try{
+                InputStream inputStream = new FileInputStream(android_linker);
+                byte[] bytes = new byte[20];
+                inputStream.read(bytes);
+                if(bytes[18] == (byte)62){
+                    return "x86_64";
+                }
+                else if(bytes[18] == (byte)183){
+                    return "aarch64";
+                }
+            }catch(IOException ex) {
+                ex.printStackTrace();
+            }
+        }
+        return "unknown";
```

Comment:
Hi, I only compile for arm64 and x86_64 arch. So any other arch is not supported. Just return "unknown", and apktool will throws aapt not found error.

Reply:
Thanks. I wonder if we can use a descriptive failure here. So we can say, "arch: ____ is not supported". Otherwise the "aapt" not found error will just say, "unknown aapt not found" which doesn't help the user.

Reply:
Yes. You can throw a more descriptive failure here and quit directly.